src/Microsoft.DotNet.Interactive/KernelScheduler.cs

```diff
@@ -115,32 +117,32 @@ private void ScheduledOperationRunLoop(object _)
                 }
                 finally
                 {
-                    _currentTopLevelOperation.Value = null;
+                    _currentTopLevelOperation = default;
                     _currentlyRunningOperation = null;
                 }
             }
         }
 
         private void Run(ScheduledOperation operation)
         {
-            _currentTopLevelOperation.Value ??= operation;
+            _currentTopLevelOperation = _currentTopLevelOperation is null? operation.Value : _currentTopLevelOperation;
```

Comment:
There's a potential race condition here.

```suggestion
            _currentTopLevelOperation = _currentTopLevelOperation ?? operation.Value;
```